tap_google_search_console/streams/abstract.py

```diff
@@ -166,6 +166,17 @@ def make_payload(self, sub_type: str, start_date: str, end_date: str, stream_met
         """Creates payload for POST API Call."""
         if self.tap_stream_id == "performance_report_custom":
             self.body_params["dimensions"] = self.set_dimensions_in_payload(stream_metadata)
+            # Skip the extraction for type discover
+            # Requests for Discover cannot be grouped by device
+            if sub_type == "discover" and "device" in self.body_params["dimensions"]:
+                self.body_params["dimensions"].remove("device")
```

Comment:
Is this removing a selected field for the purpose of the request? If so, I think we should at a minimum write a log line that indicates that we're doing this.

Reply:
This isn't removing the selected field. So, for custom reports we consider all the dimensions(page, country, device, query, date) where as for `performance_report_device` we consider only `device` as dimension.

If user selects the device field in `performance_report_custom` and while replicating data for discover `sub_type` we end up encountering the error `Requests for Discover cannot be grouped by device`, to avoid this we simple remove `device` dimension from dimensions_list for discover sub_type. and this dimension_list is passed as param in API request

Reply:
this is very similar to not considering discover sub_type for performance_report_device stream [here](https://github.com/singer-io/tap-google-search-console/blob/TDL-16008-add-new-search-types/tap_google_search_console/streams/performance_reports.py#L47)